### PR TITLE
[String] wrap(): test and fix

### PR DESCRIPTION
**Pull request by gharlan**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | n/a
| License       | MIT
| Doc PR        | symfony/symfony-docs/pull/12440

### CODE REVIEW

File: src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php [hunk 1]
```diff
@@ -11,11 +11,18 @@
 
 abstract class AbstractAsciiTestCase extends TestCase
 {
-    protected static function createFromString(string $string): AbstractString
+    protected static function stringClass(): string
```
Comment (nicolas-grekas) [inline src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php]: Is this method needed? If not, I'm not sure it's better as it creates more indirections.

**Review by nicolas-grekas — COMMENTED** [review]
Thanks, just one note.

File: src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php (hunk 1 above)
Comment (gharlan) [reply src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php]: It is used in `testWrap` for the static call.

Comment (gharlan) [reply src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php]: I could use `$class = get_class(static::createFromString(''))` instead.

Comment (nicolas-grekas) [reply src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php]: Or directly: `$s = static::createFromString(''); $s::wrap(...);`

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @gharlan.

**Pull request merged**